storage/local/frankenstein.go

```diff
@@ -273,32 +310,56 @@ func (i *Ingestor) Stop() {
 }
 
 func (i *Ingestor) loop() {
-	defer i.flushAllSeries(true)
+	defer i.flushAllUsers(true)
 	defer close(i.done)
 	tick := time.Tick(flushCheckPeriod)
 	for {
 		select {
 		case <-tick:
-			i.flushAllSeries(false)
+			i.flushAllUsers(false)
 		case <-i.quit:
 			return
 		}
 	}
 }
 
-func (i *Ingestor) flushAllSeries(immediate bool) {
+func (i *Ingestor) flushAllUsers(immediate bool) {
 	if i.chunkStore == nil {
 		return
 	}
-	for pair := range i.fpToSeries.iter() {
-		if err := i.flushSeries(pair.fp, pair.series, immediate); err != nil {
+
+	userIDs := []string{}
+	i.userStateLock.Lock()
+	for userID := range i.userState {
+		userIDs = append(userIDs, userID)
+	}
+	i.userStateLock.Unlock()
+
+	for _, userID := range userIDs {
+		i.userStateLock.Lock()
+		userState, ok := i.userState[userID]
+		i.userStateLock.Unlock()
+
+		// This should happen, right?
```

Comment:
Hmm, right now you never delete users from the user state map. So right now I think this cannot happen.